Tugas P2-Data Preprocessing | Aurelia Priscilia Simamora - 211402091

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn import preprocessing

Kode Python untuk mengimpor lima library yang umum.

# Data Formatting

## Importing Data

In [2]:
df = pd.read_csv('HousingData.csv')

adalah kode untuk membaca file CSV bernama HousingData.csv dan menyimpannya ke dalam variabel df

In [3]:
df

CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD  TAX  \
0    0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900    1  296   
1    0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671    2  242   
2    0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671    2  242   
3    0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622    3  222   
4    0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622    3  222   
..       ...   ...    ...   ...    ...    ...   ...     ...  ...  ...   
501  0.06263   0.0  11.93   0.0  0.573  6.593  69.1  2.4786    1  273   
502  0.04527   0.0  11.93   0.0  0.573  6.120  76.7  2.2875    1  273   
503  0.06076   0.0  11.93   0.0  0.573  6.976  91.0  2.1675    1  273   
504  0.10959   0.0  11.93   0.0  0.573  6.794  89.3  2.3889    1  273   
505  0.04741   0.0  11.93   0.0  0.573  6.030   NaN  2.5050    1  273   

     PTRATIO       B  LSTAT  MEDV  
0       15.3  396.90   4.98  24.0  
1       17.8  396.90   9.14  21.6  
2       17.8  392.83   4.03  34.7  
3       18.7  394.63   2.94  33.4  
4       18.7  396.90    NaN  36.2  
..       ...     ...    ...   ...  
501     21.0  391.99    NaN  22.4  
502     21.0  396.90   9.08  20.6  
503     21.0  396.90   5.64  23.9  
504     21.0  393.45   6.48  22.0  
505     21.0  396.90   7.88  11.9  

[506 rows x 14 columns]

adalah objek yang merupakan struktur data tabular yang dapat digunakan untuk menyimpan dan menganalisis data. Output di atas menunjukkan bahwa DataFrame df memiliki 506 baris dan 14 kolom. 

## Checking Column Types

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 14 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CRIM     486 non-null    float64
 1   ZN       486 non-null    float64
 2   INDUS    486 non-null    float64
 3   CHAS     486 non-null    float64
 4   NOX      506 non-null    float64
 5   RM       506 non-null    float64
 6   AGE      486 non-null    float64
 7   DIS      506 non-null    float64
 8   RAD      506 non-null    int64  
 9   TAX      506 non-null    int64  
 10  PTRATIO  506 non-null    float64
 11  B        506 non-null    float64
 12  LSTAT    486 non-null    float64
 13  MEDV     506 non-null    float64
dtypes: float64(12), int64(2)
memory usage: 55.5 KB


adalah metode yang digunakan untuk menampilkan informasi tentang DataFrame. Output di atas menunjukkan bahwa DataFrame df memiliki 506 baris dan 14 kolom. Kolom-kolom tersebut berisi informasi tentang rumah di Boston, termasuk tingkat kejahatan, zona perumahan, konsentrasi industri, keberadaan sungai, konsentrasi nitrogen oksida, jumlah kamar, usia rumah, jarak ke pusat kota, indeks aksesibilitas, pajak harta, rasio murid-guru, proporsi orang asing, status sosial ekonomi, dan harga rumah. Juga menunjukkan bahwa tipe data dari semua kolom adalah float64, kecuali kolom RAD dan TAX yang bertipe data int64.

## Check Duplicates

In [5]:
df.duplicated().sum()

0

adalah kode untuk menghitung jumlah baris duplikat.

# Handling Missing Value

In [6]:
df.drop('CRIM', axis = 'columns', inplace = True)

adalah kode untuk menghapus kolom CRIM dari DataFrame df. Parameter axis = 'columns' menunjukkan bahwa kolom yang akan dihapus adalah kolom, bukan baris. Parameter inplace = True menunjukkan bahwa DataFrame df akan dimodifikasi secara langsung.

In [7]:
df.drop('ZN', axis = 'columns', inplace = True)

In [8]:
df.drop('INDUS', axis = 'columns', inplace = True)

In [9]:
df.drop('CHAS', axis = 'columns', inplace = True)

In [10]:
df.drop('AGE', axis = 'columns', inplace = True)

In [11]:
df.drop('LSTAT', axis = 'columns', inplace = True)

In [12]:
df.columns

Index(['NOX', 'RM', 'DIS', 'RAD', 'TAX', 'PTRATIO', 'B', 'MEDV'], dtype='object')

adalah kode untuk mendapatkan nama-nama kolom. Output di atas menunjukkan bahwa DataFrame df memiliki 8 kolom, yaitu NOX, RM, DIS, RAD, TAX, PTRATIO, B, dan MEDV.

# Data Normalization

In [13]:
df

NOX     RM     DIS  RAD  TAX  PTRATIO       B  MEDV
0    0.538  6.575  4.0900    1  296     15.3  396.90  24.0
1    0.469  6.421  4.9671    2  242     17.8  396.90  21.6
2    0.469  7.185  4.9671    2  242     17.8  392.83  34.7
3    0.458  6.998  6.0622    3  222     18.7  394.63  33.4
4    0.458  7.147  6.0622    3  222     18.7  396.90  36.2
..     ...    ...     ...  ...  ...      ...     ...   ...
501  0.573  6.593  2.4786    1  273     21.0  391.99  22.4
502  0.573  6.120  2.2875    1  273     21.0  396.90  20.6
503  0.573  6.976  2.1675    1  273     21.0  396.90  23.9
504  0.573  6.794  2.3889    1  273     21.0  393.45  22.0
505  0.573  6.030  2.5050    1  273     21.0  396.90  11.9

[506 rows x 8 columns]

In [14]:
scaler = preprocessing.StandardScaler()
columns = ['RM','DIS','RAD','TAX','PTRATIO','B','MEDV']



Kode scaler = preprocessing.StandardScaler() adalah untuk membuat objek StandardScaler dari modul preprocessing di library Scikit-learn. Objek StandardScaler digunakan untuk melakukan standardisasi data

Kode columns = ['RM','DIS','RAD','TAX','PTRATIO','B','MEDV'] adalah untuk membuat list berisi nama-nama kolom yang akan distandarisasi.

In [15]:
df[columns] = scaler.fit_transform(df[columns])

adalah untuk melakukan standardisasi data pada kolom-kolom yang ditentukan oleh list columns.

In [16]:
df

NOX        RM       DIS       RAD       TAX   PTRATIO         B  \
0    0.538  0.413672  0.140214 -0.982843 -0.666608 -1.459000  0.441052   
1    0.469  0.194274  0.557160 -0.867883 -0.987329 -0.303094  0.441052   
2    0.469  1.282714  0.557160 -0.867883 -0.987329 -0.303094  0.396427   
3    0.458  1.016303  1.077737 -0.752922 -1.106115  0.113032  0.416163   
4    0.458  1.228577  1.077737 -0.752922 -1.106115  0.113032  0.441052   
..     ...       ...       ...       ...       ...       ...       ...   
501  0.573  0.439316 -0.625796 -0.982843 -0.803212  1.176466  0.387217   
502  0.573 -0.234548 -0.716639 -0.982843 -0.803212  1.176466  0.441052   
503  0.573  0.984960 -0.773684 -0.982843 -0.803212  1.176466  0.441052   
504  0.573  0.725672 -0.668437 -0.982843 -0.803212  1.176466  0.403225   
505  0.573 -0.362767 -0.613246 -0.982843 -0.803212  1.176466  0.441052   

         MEDV  
0    0.159686  
1   -0.101524  
2    1.324247  
3    1.182758  
4    1.487503  
..        ...  
501 -0.014454  
502 -0.210362  
503  0.148802  
504 -0.057989  
505 -1.157248  

[506 rows x 8 columns]

# Data Binning

Tidak perlu karena data-datanya sudah terdistribusi dengan baik.